pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -272,6 +272,17 @@ func (g *gatewayMonitor) startControllers() error {
 
 	g.controllers = append(g.controllers, c)
 
+	// A user is not normally expected to delete the internal service created by the Globalnet controller.
+	// However, when it's accidentally done while the globalnet controller is down, the internal service
+	// remains until the finalizer is removed. We have seen that this intermediate state of
+	// the service can potentially create issues in some deployments. Hence, we identify such internal
+	// services and delete them when the Globalnet controller pod comes up.
+	err = RemoveStaleInternalServices(g.syncerConfig)
+	if err != nil {
+		// Just log an error message as its non-fatal
+		logger.Errorf(err, "Error removing stale internal services created by Globalnet controller")
```

Comment:
I know it’s not as convenient, but if this isn’t an actual error, we shouldn’t log it as an error:

```suggestion
		logger.Warningf("Error removing stale internal services created by Globalnet controller: %v", err)
```

Reply:
Some CNIs like kube-proxy continue to support the service (by load-balancing the requests to the backend pods) even if the deletionTimestamp is set on the service. However, others like OVNK do not support it. So, I'd say that it would be better to log an error message as it would be difficult to spot the issue otherwise.

Reply:
I've added a warning message if a internal service with deletionTimestamp exists. Hope you are fine with having this as an error message.